src/System.Buffers.Primitives/System/Buffers/Buffer.cs

```diff
@@ -69,24 +70,19 @@ public Buffer(T[] array, int start, int length)
             if ((uint)start > (uint)array.Length || (uint)length > (uint)(array.Length - start))
```

Comment:
> Why casting to uint is necessary?

`(uint)start > (uint)array.Length` tests both a negative `start` and `start` being larger than `Length` as a negative `int` cast to a `uint` is larger than `int.MaxValue`.

Now it is known `start` is within bounds, `(uint)length > (uint)(array.Length - start))` tests whether `length` is negative or larger than available.

So 4 comparisons are done in the statement, using 2 comparisons.

Not using the cast to `uint` the equivalent would be:

```csharp
if (
    start  < 0 || start > array.Length || 
    length < 0 || length > (array.Length - start)
   )
```